lib/Parse/ParseExpr.cpp

```diff
@@ -2399,6 +2399,39 @@ parseClosureSignatureIfPresent(SmallVectorImpl<CaptureListEntry> &captureList,
     }
   }
 
+  if (!params)
+    return invalid;
+
+  // If this was a closure declaration (maybe even trailing)
+  // it's imperative to detect any parameter list destructuring
+  // early on, see SE-0110 for more details.
+  for (unsigned i = 0, e = params->size(); i != e; ++i) {
```

Comment:
Minor nit: This makes it sound like SE-0110 has details about why it's imperative to detect parameter list destructuring. You might want to turn that comma into a period and then add a line explaining why it's important to do this here.